### ISSUE TITLE
Checks to deal with volatile memory operations should have their own functions

### ISSUE THREAD

**Issue by Leonardo2718** [eclipse/omr#225]
Currently, in `OMR::Power::TreeEvaluator`, there are several occurrences of code that check:
- if a node being evaluated requires additional `sync` instructions to be generated
- if a node deals with lazy volatiles

The code that checks for a required `sync` instruction is general and deserves to be in its own function in `OMR::TreeEvaluator`. Likewise, the code to check if a node refers to a lazy volatile is general and can be put into its own function, though the function should go in the Power specialization of `TreeEvaluator`.

Creating and using functions to perform these checks must be done in three steps

- [ ] implement the functions in OMR
- [ ] create overrides for the implementations in downstream projects
- [ ] replace occurrences of the check code in OMR with calls to the new functions

**Comment by 0xdaryl** [comment]
Good idea Leonardo.  That function should also be subject to project specialization as well because volatile semantics may be different depending on the memory model of the target language.
